### PR TITLE
Partiall implementation of `sysinfo` syscall (`uptime` calculation)

### PR DESCRIPTION
**Pull request by c71n93**
This PR partially implements `sysinfo` syscall with `sysinfo` struct. Only `uptime` calculation was implemented. This fixes the `uptime` tool, which previously printed random information.

- before:
Screenshot from 2024-12-11 23-45-32

- after
Screenshot from 2024-12-11 23-32-01

Many thanks to @levBagryansky

### CODE REVIEW

File: kernel/src/syscall/sysinfo.rs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    freeswap: u64,
+    procs: u16,
+    totalhigh: u64,
+    freehigh: u64,
+    mem_unit: u32,
+}
+
+pub fn sys_sysinfo(sysinfo_addr: Vaddr, ctx: &Context) -> Result<SyscallReturn> {
+    let info = sysinfo {
+        uptime: read_monotonic_time().as_secs() as i64,
```
Comment (grief8) [inline kernel/src/syscall/sysinfo.rs:27]: I think we can add more information here. 
https://github.com/asterinas/asterinas/blob/4316247d82d86e1230f23883c193c26133b12e6a/ostd/src/mm/stat/mod.rs#L12-L21

**Review by grief8 — COMMENTED** [review]
Welcome to Asterinas! The demand of `sysinfo` is also mentioned in #1681. I just have a minor suggestion.

**Comment by tatetian** [comment]
@c71n93 Thanks for adding this new system call! Could you also update the [Linux compatibility page of the Asterinas Book](https://github.com/asterinas/asterinas/blob/main/docs/src/kernel/linux-compatibility.md)?

**Comment by c71n93** [comment]
> @c71n93 Thanks for adding this new system call! Could you also update the [Linux compatibility page of the Asterinas Book](https://github.com/asterinas/asterinas/blob/main/docs/src/kernel/linux-compatibility.md)?

@tatetian done!

File: kernel/src/syscall/sysinfo.rs (hunk 1 above)
Comment (c71n93) [reply kernel/src/syscall/sysinfo.rs:28]: @grief8 Added `totalram` and `freeram`. 

However integration tests `syscall_test_at_exfat_linux` and `smp_syscall_test_mb2` are failed. And this tests failed extremely strange as for me:
- `syscall_test_at_exfat_linux` failed only at test case `MMapTest.NoReserve`, which is not related to this PR (at least directly).
- `smp_syscall_test_mb2` failed with panic while starting `mknod_test` (according to logs) without even starting `SysinfoTest`.

I don't have enough knowledge about Asterinas and its test system to quickly figure out what's gone wrong. And all syscall tests are passed on my machine (as well as in other workflows, e.g. [this](https://github.com/asterinas/asterinas/actions/runs/12305052400/job/34343819474?pr=1698)) so it becomes hard to debug this failures. Could you please explain what is the idea of different syscalls tests?

Comment (grief8) [reply kernel/src/syscall/sysinfo.rs:28]: Never mind. This failure has nothing to do with this pull request. It is a random bug that is being fixed. We can rerun the workflow with the help of `boterinas` by commenting `@boterinas rerun` to solve it.

Comment (grief8) [reply kernel/src/syscall/sysinfo.rs:28]: @boterinas rerun

**Comment by tatetian** [comment]
@boterinas rerun

File: kernel/src/syscall/sysinfo.rs (hunk 1 above)
Comment (c71n93) [reply kernel/src/syscall/sysinfo.rs:28]: @grief8 This comment is resolved then?

Comment (grief8) [reply kernel/src/syscall/sysinfo.rs:28]: All right.

**Comment by c71n93** [comment]
So, @tatetian @grief8 WDYT?

**Review by grief8 — APPROVED** [review]
LGTM!

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**